static/js/typeahead_helper.js

```diff
@@ -207,17 +207,27 @@ exports.compare_by_sub_count = function (stream_a, stream_b) {
     return stream_a.subscribers.num_items() < stream_b.subscribers.num_items();
 };
 
+// Find which stream comes first in the sidebar listing order
+exports.compare_by_sidebar = function (stream_a, stream_b) {
+    var a_score = stream_data.is_active(stream_a.name) + stream_a.pin_to_top * 2;
+    var b_score = stream_data.is_active(stream_b.name) + stream_b.pin_to_top * 2;
```

Comment:
For readability, I'd probably spell this `2 * stream_a.pin_to_top + stream_data.is_active(stream_a.name)`

That said, I'm not a fan of implicitly treating bools and ints in this sort of context, so it might be better to extract a `score` function.

(As a sidenote, it's theoretically possible to reuse code from `stream_sort.js`, but I'm not sure it's worth it.).